src/detect-mark.c

```diff
@@ -253,81 +253,65 @@ static int DetectMarkPacket(DetectEngineThreadCtx *det_ctx, Packet *p,
 /**
  * \test MarkTestParse01 is a test for a valid mark value
  *
- *  \retval 1 on succces
- *  \retval 0 on failure
  */
 static int MarkTestParse01 (void)
 {
     DetectMarkData *data;
 
     data = DetectMarkParse("1/1");
 
-    if (data == NULL) {
-        return 0;
-    }
+    FAIL_IF_NULL(data);
 
     DetectMarkDataFree(NULL, data);
-    return 1;
+    PASS;
 }
 
 /**
  * \test MarkTestParse02 is a test for an invalid mark value
  *
- *  \retval 1 on succces
- *  \retval 0 on failure
  */
 static int MarkTestParse02 (void)
 {
     DetectMarkData *data;
 
     data = DetectMarkParse("4");
 
-    if (data == NULL) {
-        return 1;
-    }
+    PASS_IF(data == NULL);
```

Comment:
@victorjulien do we need a `PASS_IF_NULL` macro?

Reply:
That would be a nice addition.

Reply:
Creating an Outreachy ticket for that then.

Reply:
> Creating an Outreachy ticket for that then.

If such a ticket is created, may I take it up after asking in the mailing list

Reply:
Sure. It has already been created. https://redmine.openinfosecfoundation.org/issues/4066

Reply:
Hello,
Shall I incorporate the PASS_IF_NULL macro in detect-mark.c too?

Reply:
Yes, please. After its done.

Reply:
Shall I remove the description for the macro in #5485 and do the above task in a single commit by opening a new PR?

Reply:
Since the addition of macro is not the same as converting unittest to PASS/FAIL API, you should make two commits, first for the addition of the new macro and second to convert unittests to PASS/FAIL API using that macro. You can do this as a part of the same PR.
Make sure each commit individually passes the build test. (e.g. if you do the conversion first and then add macro later, you'll see the same failure in travis that you did in the first version of your PR)